Add tests for color.space()

SondraE opened the pull request:
![](https://images.unsplash.com/photo-1497781495506-ce58b286d8f5?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwxfDB8MXxyYW5kb218MHx8Y3V0ZSxhbmltYWwsQ0hBTkdFX01FfHx8fHx8MTY3MDM1NjkxOQ&ixlib=rb-4.0.3&q=80&)


## Steps to test/reproduce
- new file space.hrx
- tests and errors

SondraE committed:
added tests and errors for color-space

SondraE committed:
renamed file space

stacyk commented on spec/core_functions/color/space.hrx:
```diff
@@ -0,0 +1,234 @@
+@use "sass:color";
+a {b: color.space(#f2ece4)}
+
+<===> hex/output.css
+a {
+  b: #f2ece4;
+}
```
@dvdherron  @SondraE I would have expected RGB for this one?

stacyk replied:
https://css.oddbird.net/sass/color-spaces/proposal/#known-color-space

> "Colors specified using a CSS color keyword or the hex notation are converted to rgb and serialized as part of the rgb color space."

dvdherron replied:
yup, rgb

stacyk commented on spec/core_functions/color/space.hrx:
```diff
@@ -0,0 +1,234 @@
+
+<===> named/output.css
+a {
+  b: midnightblue;
```
same here, I would have thought RGB?

dvdherron replied:
named colors are in the rgb space as well.

stacyk commented on spec/core_functions/color/space.hrx:
```diff
@@ -0,0 +1,234 @@
+
+<===> rgba/output.css
+a {
+  b: rgba;
```
@SondraE I'd need someone else to confirm, but I'm not sure the alpha channel makes it a different space? @dvdherron?

dvdherron replied:
correct. This would still be in the rgb space.

stacyk reviewed (commented):
@SondraE Looking good! I left a few comments, but think @dvdherron or @mirisuzanne should confirm a few of these before you adjust anything.

dvdherron commented on spec/core_functions/color/space.hrx:
```diff
@@ -0,0 +1,234 @@
+<===>
+================================================================================
+<===> error/too_few_args/input.scss
+a {b: color.space(rgb(0 255))}
```
Since this function only takes one argument (`$color`), too few arguments would be an empty function `color.space()`.

dvdherron commented on spec/core_functions/color/space.hrx:
```diff
@@ -0,0 +1,234 @@
+<===>
+================================================================================
+<===> error/too_many_args/input.scss
+a {b: color.space(rgb(0 255 0), blue)}
```
``
```suggestion
a {b: color.space(rgb(0 255 0), a)}
```

I think the styleguide says that irrelevant placeholders should be single-letter names.

SondraE replied:
Okay. I should probably go back to my is-legacy PR and update that as well.

dvdherron commented on spec/core_functions/color/space.hrx:
```diff
@@ -0,0 +1,234 @@
+
+<===>
+================================================================================
+<===> error/too_few_args/input.scss
```
```suggestion
<===> error/too_few_args/input.scss
@use "sass:color";
```

SondraE replied:
I copied and pasted my is-legacy tests, so it looks like this is something I should add to is-legacy pr as well.

dvdherron commented on spec/core_functions/color/space.hrx:
```diff
@@ -0,0 +1,234 @@
+
+<===>
+================================================================================
+<===> error/too_many_args/input.scss
```
```suggestion
<===> error/too_many_args/input.scss
@use "sass:color";
```

dvdherron commented on spec/core_functions/color/space.hrx:
```diff
@@ -0,0 +1,234 @@
+<===>
+================================================================================
+<===> error/type/input.scss
+a {b: color.space(1)}
```
```suggestion
@use "sass:color";
a {b: color.space(1)}
```

dvdherron reviewed (commented):
some minor cleanup but this is looking good!

SondraE committed:
updated rgb space and fixed errors

stacyk reviewed (approved):
Looks good to me  :)

nex3 committed:
[Color 4] Update tests for Color Level 4 (#1846)

This updates most tests for the currently-implemented chunk of the
Color 4 feature. A few tests that need more thorough reworking (such
as being moved out of `error/` for things that are no longer intended
to be errors) are marked as "todo" instead.

See #1828
See sass/sass#2831

dvdherron committed:
Merge branch 'feature.color-4' into color-space

nex3 committed:
Update specs to match Dart Sass
